File: Solutions/104- Maximum Depth of Binary Tree/Solution.java
class Solution {
    public int maxDepth(TreeNode root) {
        if (root == null) {
            return 0;
        }

        int depth_right = maxDepth(root.right);
        int depth_left = maxDepth((root.left));

        return Math.max(depth_left,depth_right) + 1;
    }
}

File: Solutions/104- Maximum Depth of Binary Tree/Solution.py
class Solution:
    def maxDepth(self, root: Optional[TreeNode]) -> int:
        if root is None:
            return 0

        depth_left = self.maxDepth(root.left)
        depth_right = self.maxDepth(root.right)

        return max(depth_left,depth_right) + 1